From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jbicha@ubuntu.com>
Date: Sun, 9 Aug 2026 14:16:00 +0200
Subject: Revert "datetime: Use GtkEnumList"

Revert "datetime: disable ellipzation for one AdwComboRow"
Revert "build: Bump GTK dependency"

This reverts commit b186818148fb85e48c9d35ae0861b7e27e61c42c.
This reverts commit 0310070574837b66f22e509947acf665281da089.
This revert commit 7b741d9374fb2ffd4a5fc8ae58c88caec2862d13.

Debian Unstable does not yet have gtk 4.23
---
 meson.build                                 |  2 +-
 panels/system/datetime/cc-datetime-page.blp |  4 ++-
 panels/system/datetime/cc-datetime-page.c   | 41 +++--------------------------
 3 files changed, 7 insertions(+), 40 deletions(-)

diff --git a/meson.build b/meson.build
index 49fca1b..ef4ea76 100644
--- a/meson.build
+++ b/meson.build
@@ -129,7 +129,7 @@ libgvc = subproject(
 libgvc_dep = libgvc.get_variable('libgvc_dep')
 
 libadwaita_version = '>= 1.8.alpha'
-gtk_version = '>= 4.23.0'
+gtk_version = '>= 4.22.0'
 goa_req_version = '>= 3.51.0'
 pulse_req_version = '>= 2.0'
 malcontent_version = '>= 0.13'
diff --git a/panels/system/datetime/cc-datetime-page.blp b/panels/system/datetime/cc-datetime-page.blp
index 9d72dc2..6802ec1 100644
--- a/panels/system/datetime/cc-datetime-page.blp
+++ b/panels/system/datetime/cc-datetime-page.blp
@@ -156,7 +156,9 @@ template $CcDateTimePage: Adw.NavigationPage {
           Adw.ComboRow week_start_day_row {
             title: _("F_irst Day of the Week");
             use-underline: true;
-            model: EnumList {
+            expression: expr $get_weekday_name() as <string>;
+
+            model: Adw.EnumListModel {
               enum-type: typeof<$GDesktopWeekday>;
             };
           }
diff --git a/panels/system/datetime/cc-datetime-page.c b/panels/system/datetime/cc-datetime-page.c
index 7768541..2e28060 100644
--- a/panels/system/datetime/cc-datetime-page.c
+++ b/panels/system/datetime/cc-datetime-page.c
@@ -209,11 +209,11 @@ static void on_month_selection_changed_cb (CcDateTimePage *self);
 static void time_changed_cb (CcDateTimePage *self, CcTimeEditor *editor);
 
 static gchar *
-get_weekday_name (GtkEnumListItem *item)
+get_weekday_name (AdwEnumListItem *item, gpointer user_data)
 {
     gint selected_value;
 
-    selected_value = gtk_enum_list_item_get_value (item);
+    selected_value = adw_enum_list_item_get_value (item);
     if (selected_value == 0)
         return g_strdup (_("Locale Default"));
 
@@ -761,6 +761,7 @@ cc_date_time_page_class_init (CcDateTimePageClass *klass)
     gtk_widget_class_bind_template_callback (widget_class, list_box_row_activated);
     gtk_widget_class_bind_template_callback (widget_class, change_clock_settings_cb);
     gtk_widget_class_bind_template_callback (widget_class, on_date_box_row_activated_cb);
+    gtk_widget_class_bind_template_callback (widget_class, get_weekday_name);
 
     bind_textdomain_codeset (GETTEXT_PACKAGE_TIMEZONES, "UTF-8");
 }
@@ -795,34 +796,10 @@ get_week_start_day_setting (GValue *property_value, GVariant *setting_source, gp
     return FALSE;
 }
 
-static void
-on_factory_setup_week_start_day (GtkSignalListItemFactory *factory, GtkListItem *item, gpointer user_data)
-{
-    GtkWidget *label = gtk_label_new (NULL);
-    gtk_label_set_ellipsize (GTK_LABEL (label), PANGO_ELLIPSIZE_NONE);
-    gtk_label_set_xalign (GTK_LABEL (label), 0.0);
-    gtk_list_item_set_child (item, label);
-}
-
-static void
-on_factory_bind_week_start_day (GtkSignalListItemFactory *factory, GtkListItem *item, gpointer user_data)
-{
-    GtkLabel *label = GTK_LABEL (gtk_list_item_get_child (item));
-    GtkEnumListItem *enum_item = GTK_ENUM_LIST_ITEM (gtk_list_item_get_item (item));
-    gchar *text = get_weekday_name (enum_item);
-
-    if (text) {
-        gtk_label_set_text (label, text);
-        g_free (text);
-    }
-}
-
 static void
 cc_date_time_page_init (CcDateTimePage *self)
 {
     g_autoptr(GError) error = NULL;
-    g_autoptr (GtkListItemFactory) factory = NULL;
-    g_autoptr (GtkListItemFactory) list_factory = NULL;
 
     gtk_widget_init_template (GTK_WIDGET (self));
 
@@ -942,16 +919,4 @@ cc_date_time_page_init (CcDateTimePage *self)
     /* We ignore UTC <--> LocalRTC changes at the moment */
 
     self->filechooser_settings = g_settings_new (FILECHOOSER_SCHEMA);
-
-    /* Setup custom factories for AdwComboRow week_start_day_row to avoid default ellipzation #3735 */
-    factory = gtk_signal_list_item_factory_new ();
-    g_signal_connect (factory, "setup", G_CALLBACK (on_factory_setup_week_start_day), NULL);
-    g_signal_connect (factory, "bind", G_CALLBACK (on_factory_bind_week_start_day), NULL);
-
-    list_factory = gtk_signal_list_item_factory_new ();
-    g_signal_connect (list_factory, "setup", G_CALLBACK (on_factory_setup_week_start_day), NULL);
-    g_signal_connect (list_factory, "bind", G_CALLBACK (on_factory_bind_week_start_day), NULL);
-
-    adw_combo_row_set_factory (self->week_start_day_row, factory);
-    adw_combo_row_set_list_factory (self->week_start_day_row, list_factory);
 }
